service/storage_service.cc

```diff
@@ -2878,7 +2878,7 @@ future<> storage_service::replicate_to_all_cores(mutable_token_metadata_ptr tmpt
             auto& table_erms = pending_table_erms[this_shard_id()];
             for (auto it = table_erms.begin(); it != table_erms.end(); ) {
                 auto& cf = db.find_column_family(it->first);
-                cf.update_effective_replication_map(std::move(it->second));
+                co_await cf.update_effective_replication_map(std::move(it->second));
```

Comment:
This entire function `storage_service::replicate_to_all_cores` tries very hard to make updates atomic/isolated i.e. that every shard observes all cf/erm changes at a single instant. It goes a long way to construct a `pending_table_erms` struct and so on, so it can then move those erms to the observable state in a single task without a yield.

And then this simple `co_await` abandons all of that work. Breaking the purpose of the multi-phase approach that this function took.

So now we can end up in a state, for example, when we have a table with materialized view attached to it, where the base erm reflects a topology change, while view erm still doesn't -- or vice versa. And data path queries will observe such inconsistent state.